test/Table_Tests/src/Common_Table_Operations/Take_Drop_Spec.enso

```diff
@@ -234,6 +239,14 @@ spec setup =
             alpha.drop (Last -1) . should_equal alpha
             alpha.drop (Last 100) . should_equal empty_alpha
 
+        Test.specify "should handle consecutive take/drops" <|
+            alpha.take 5 . sort . take 3 . to_vector . should_equal [1, 2, 3]
+            alpha.take 3 . sort . take 5 . to_vector . should_equal [1, 2, 3]
+            alpha.take 5 . sort . drop 3 . to_vector . should_equal [4, 5]
+            alpha.drop 3 . sort . drop 2 . to_vector . should_equal [6, 7, 8]
+            alpha.drop 2 . sort . drop 3 . to_vector . should_equal [6, 7, 8]
+            alpha.drop 3 . sort . take 2 . to_vector . should_equal [4, 5]
```

Comment:
Take requires an ordering on its input, so the sort is necessary.

Reply:
Hmmmm, I see, but that is concerning then. IMO two `take`s should be possible without a sort in-between.

I think I get why this doesn't work, but in principle - it should. If I have a sorted collection and take/drop from it - it is _still_ sorted.

I think we should amend our `default_ordering` logic to be able to still detect an ordering after performing `take`/`drop`. We may need to have some additional flag on the context specifying an 'inherited ordering' (as we should not re-specify the normal ordering which would cause 2 levels of nested subqueries to both have `ORDER BY` clauses).
What do you think?